client-core/src/service/key_service.rs

```diff
@@ -21,32 +21,32 @@ where
         KeyService { storage }
     }
 
-    /// Adds a new public-private keypair to storage
-    pub fn add_keypair(
+    /// Adds a new wallet_name-private keypair to storage
+    pub fn add_wallet_private_key(
         &self,
+        wallet_name: &str,
         private_key: &PrivateKey,
-        public_key: &PublicKey,
         enckey: &SecKey,
     ) -> Result<()> {
         self.storage
             .set_secure(
                 KEYSPACE,
-                public_key.serialize(),
+                wallet_name.as_bytes(),
                 private_key.serialize(),
                 enckey,
             )
             .map(|_| ())
     }
 
-    /// Retrieves private key corresponding to given public key
-    pub fn private_key(
+    /// Retrieves private key corresponding to given wallet name
```

Comment:
one wallet, one privatekey,
1:1 right?

Reply:
yes, the private key is the one when the wallet create, so there is only one private key.